29-restroom-servicing-monitor/README.md

```diff
@@ -0,0 +1,98 @@
+
+## Sparrow Setup
+
+### Quickstart
+
+Follow the [Sparrow Quickstart](https://dev.blues.io/quickstart/sparrow-quickstart/) to get your Sparrow devices paired with each other and associated with the Notehub project you just created. Make sure that you use the ProductUID generated in [Notehub Setup](#notehub-setup) when it comes time to issue the `hub.set` command in the quickstart. Note also that we'll only need one reference node for this project, so you don't need to pair both nodes that came with the dev kit.
+
+After you've completed the quickstart, leave the Notecarrier and essentials board powered and connected. These two devices will act as our gateway to Notehub, and we won't need to touch them again. The rest of this guide will focus on the reference node.
+
+### Hardware
+
+Before we set up the custom firmware needed for the restroom servicing monitor, let's wire up the hardware. The hardware interface is 3 push buttons: 1 for a bad rating (restroom needs service), 1 for an ok rating (restroom cleanliness deteriorating), and 1 for a good rating (no service needed). These buttons are wired to GPIO pins on the reference node. Those pins use internal pull-ups so that the pin reads a digital 1 when the button isn't pushed. When the button is pushed, it connects the corresponding GPIO pin to GND so that it reads a digital 0. Using the breadboard, buttons, and jumper wires, wire up the hardware as shown in the diagram below.
+
+![breadboard wiring](assets/nf29_breadboard.jpg "Breadboard Wiring")
+
+The labels off the breadboard (A1-A3 and GND) correspond to pins on the reference node. Connect these to the breadboard as shown with jumper wires.
+
+### Firmware
+
+Next, we need to flash the reference node with the restroom servicing monitor firmware.
+
+1. Before we do anything, we need to pull in some dependencies for the firmware to work. After cloning this repository, run `git submodule update --init`. This will pull in the note-c and sparrow-lora submodules that this project depends on.
+1. Install [Visual Studio Code](https://code.visualstudio.com/). We will program the reference node from VS Code.
+1. Add the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) to VS Code. Building and flashing the firmware will take place inside a Docker container specifically designed for these tasks (refer to firmware/Dockerfile for details of what's in the container).
+1. Connect the STLINK-V3MINI to your development PC with a USB A to micro USB cable.
+1. Connect the STLINK to your reference node with the 2x7 JTAG ribbon cable.
+1. Navigate to File > Open Folder and select the firmware folder.
+1. In the bottom left corner, click the lavender icon (it looks like "><") and hit Reopen in Container. The first time you do this, the Docker container will have to be built, which will take several minutes.
```

Comment:
Dev Containers will work on both Mac and Windows, but you are constrained to "building only". Both flashing and debugging require USB access - which is native to Linux, but doesn't work well on Windows and Mac, at this time.

Reply:
Noted.

Reply:
We could use platform IO for that. They have "remote embedded development". Run the server on your host, and the client in docker so that it can flash to the host machine.

Reply:
No work has been done to determine if Platform.io could support Sparrow, nor the work required to support Sparrow.